lib/Connection.php

```diff
@@ -86,35 +86,45 @@
 	 *   A connection name that is set in ActiveRecord\Config
 	 *   If null it will use the default connection specified by ActiveRecord\Config->set_default_connection
 	 * @return Connection
-	 * @see parse_connection_url
+	 * @see ConnectionInfo::from_connection_url
 	 */
-	public static function instance($connection_string_or_connection_name=null)
+	public static function instance($connection_info_or_name=null)
 	{
 		$config = Config::instance();
 
-		if (strpos($connection_string_or_connection_name, '://') === false)
+		if (!is_array($connection_info_or_name))
 		{
-			$connection_string = $connection_string_or_connection_name ?
-				$config->get_connection($connection_string_or_connection_name) :
-				$config->get_default_connection_string();
+			if (strpos($connection_info_or_name, '://') === false)
+			{
+				$connection_url = $connection_info_or_name ?
+					$config->get_connection($connection_info_or_name) :
+					$config->get_default_connection_string();
+			}
+			else
+			{
+				$connection_url = $connection_info_or_name;
+			}
+
+			if (!$connection_url)
+				throw new DatabaseException("Empty connection string");
+
+			$connection_info = ConnectionInfo::from_connection_url($connection_url);
 		}
 		else
-			$connection_string = $connection_string_or_connection_name;
-
-		if (!$connection_string)
-			throw new DatabaseException("Empty connection string");
+		{
+			$connection_info = new ConnectionInfo($connection_info_or_name);
+		}
```

Comment:
Why not add in a block that checks if the `$connection_info_or_name` is an instance of the `ConnectionInfo` class? That way you can pass around the value object directly and it doesn't have to be re-built each time.